kani-compiler/src/codegen_cprover_gotoc/utils/utils.rs

```diff
@@ -61,6 +42,30 @@ impl<'tcx> GotocCtx<'tcx> {
         }
         s
     }
+
+    /// Tries to extract a string message from an `Expr`.  If the expression is
+    /// a pointer to a variable that represents a string literal (as created in
+    /// codegen_slice_value), this will return the string constant. Otherwise,
+    /// return `None`.
+    pub fn extract_const_message(&self, arg: &Expr) -> Option<String> {
+        match arg.value() {
+            ExprValue::Struct { values } => match &values[0].value() {
+                ExprValue::Typecast(expr) => match expr.value() {
+                    ExprValue::AddressOf(address) => match address.value() {
+                        ExprValue::Symbol { identifier } => {
+                            // lookup the string literal in the goto context
+                            let name = self.str_literals.get(identifier).unwrap();
```

Comment:
not for this PR, but I was wondering if we could map directly Expr -> String Literal.

Reply:
Yes, that would be better. I couldn't do that because `Expr` is not "hash"able. I'll add a comment though.